### ISSUE TITLE
Saving trace in a temp file or RAM

### ISSUE THREAD

**Issue by oleedd** [x64dbg/x64dbg#3041]
### Feature type

Debugging

### Describe the feature

How to make it use a temp file or RAM (better) for tracing? It always asks about a file, it annoys very much. If someone needs it in the future, he could save it (ctrl+s).

**Comment by torusrxxx** [comment]
It indeed requires a file to save data into. Using a temp file in the system temporary directory could work. Right now you can also use the "close and delete" context menu to quickly delete the trace file if you don't need it afterwards.

**Comment by mrexodia** [comment]
Yeah I think the easiest solution is a checkbox “don’t prompt again” and just create the file automatically in the “db” folder. The user can always create a custom file manually.

**Comment by oleedd** [comment]
Better to do in the way it works in OllyDbg: to use RAM of a default temp file (a static name), and if a user wants to save it, to ask him about the name. You save all traces now.

**Comment by mrexodia** [comment]
Again, you are welcome to contribute and implement it as you see fit.

**Comment by oleedd** [comment]
I can't code this for now( That is why I created an issue, not a PR.
So:
1\. As I see, it doesn't have a button for saving. Adding it.
2\. Removing promt and using a default temp name.
That is all to be perfect.

**Comment by oleedd** [comment]
I can't code this for now( That is why I created an issue, not a PR.
So:
1\. As I see, it doesn't have a button for saving. Adding it. That prompt might be used for this.
2\. Using a default temp name instead of asking.
That is all to be perfect.

**Comment by oleedd** [comment]
@mrexodia Please solve this. Using RAM is the best.

**Comment by mrexodia** [comment]
You can create the trace on a RAM drive
